java_src/foam/nanos/logger/NanoLogger.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.logger;
+
+import foam.core.*;
+import foam.nanos.NanoService;
+import java.util.logging.*;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.io.IOException;
+
+public class NanoLogger extends ContextAwareSupport implements NanoService {
+  private static final SimpleDateFormat sdf = new SimpleDateFormat("yyyy.MM.dd.HH.mm.ss");
+  Logger logger;
+  StringBuffer sb;
+
+  public void start() {
+    logger = Logger.getAnonymousLogger();
+    logger.setUseParentHandlers(false);
+    sb = new StringBuffer();
+    try {
+      Handler handler = new FileHandler("nano.log");
+      handler.setFormatter(new CustomFormatter());
+      logger.addHandler(handler);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private class CustomFormatter extends Formatter {
+    long prevTime;
+    String prevTimestamp;
+    @Override
+    public String format(LogRecord record) {
+      if (prevTime / 1000 != System.currentTimeMillis() / 1000) {
+        prevTime = System.currentTimeMillis();
+        prevTimestamp = sdf.format(new Timestamp(prevTime));
+      }
+      sb.setLength(0);
+      int lev = record.getLevel().intValue();
+      String msg = record.getMessage();
+
+      sb.append(prevTimestamp);
+      sb.append(',');
+
+      // debug special case
+      if ((lev == 900) && (msg.substring(0, Math.min(msg.length(), 7)).equals("~DEBUG~"))) {
+        sb.append("DEBUG");
+        msg = msg.substring(7, msg.length());
+      } else {
+        sb.append(record.getLevel());
+      }
+
+      sb.append(',');
+      sb.append(msg);
+      sb.append('\n');
+      return sb.toString();
+    }
+  }
+
+  public void log(String msg) {
+    logger.info(msg);
+  }
+
+  public void info(String msg) {
+    logger.info(msg);
+  }
+
+  public void warning(String msg) {
+    logger.warning(msg);
+  }
+
+  public void error(String msg) {
+    logger.severe(msg);
+  }
+
+  // can't normally do .debug() with custom formatter: append prefix to indicate debug
+  public void debug(String msg) {
+    logger.warning("~DEBUG~" + msg);
```

Comment:
I couldn't find any way to implement that using the custom formatter. the custom formatter only allows a single variable (msg) the be passed. The 3 I used are the only ones that it implements as well.